lib/utils/cobs.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2024 Kelly Helmut Lord
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/net_buf.h>
+
+#include <zephyr/data/cobs.h>
+
+/**
+ * @brief COBS encode data from a net_buf into another net_buf
+ *
+ * @param src Source net_buf containing data to encode
+ * @param dst Destination net_buf to store encoded data
+ * @return int 0 on success, negative errno on failure
+ */
+int cobs_encode(struct net_buf *src, struct net_buf *dst)
+{
+	/* Calculate required space for worst case */
+	size_t max_encoded_size = src->len + (src->len / 254) + 2;
+
+	/* Check if destination has enough space */
+	if (net_buf_tailroom(dst) < max_encoded_size) {
+		return -ENOSPC;
+	}
+
+	uint8_t *code_ptr = net_buf_add(dst, 1);
+	uint8_t code = 1;
+
+	/* Process all input bytes */
+	for (size_t i = 0; i < src->len; i++) {
+		if (src->data[i] == 0) {
```

Comment:
Similar to adding to the destination buffer, I'd pull from the source buffer.

```suggestion
	uint8_t data;
	while (src->len > 0) {
		data = net_buf_pull_u8(src);

		if (data == 0) {
```